cdap-ui/server/url-helper.js

```diff
@@ -19,31 +19,63 @@
 const REQUEST_ORIGIN_ROUTER = 'ROUTER';
 const REQUEST_ORIGIN_MARKET = 'MARKET';
 
+// return market host string from path's query parameter. empty string will be
+// return if not found.
+function getMarketHostParam(path) {
+  const EXTRACT_MARKET_HOST_REGEX = /[?|&]marketHost=([^&]+).*$/;
+  const matches = path.match(EXTRACT_MARKET_HOST_REGEX);
+  return matches ? matches[1] : '';
+}
+
+function getMarketUrls(cdapConfig) {
+  return cdapConfig['market.base.urls'] ? cdapConfig['market.base.urls'].split('+') : [];
+}
+
+function getVerifiedMarketHost(cdapConfig, urlOrPath) {
+    if (!cdapConfig['market.base.urls']) {
+      return `${cdapConfig['market.base.url']}`;
+    }
+
+    const marketHost = getMarketHostParam(urlOrPath);
+    const found = getMarketUrls(cdapConfig).find(element => element === decodeURIComponent(marketHost));
+    return found ? found : '';
```

Comment:
This should check if the sourceLink is a valid market url and return undefined. 
The consumer (`/forwardMarketToCdap`) API should send a 403 response denying the request.

As an attacker I can construct a link like https://<cdap-instance>/forwardMarketToCdap?sourceLink="mymaliciousdomain.com/packages/malware"&targetLink="cdap-instance"